services/m3dbnode/main/main.go

```diff
@@ -83,11 +84,20 @@ func main() {
 			dbOpts,
 			doneCh,
 		); err != nil {
-			log.Fatalf("serve error: %v", err)
+			log.Fatalf("server fatal error: %v", err)
 		}
+
+		// Server is closed
+		<-closedCh
 	}()
 
-	log.Fatalf("interrupt: %v", interrupt())
+	// Handle interrupt
+	log.Infof("interrupt: %v", interrupt())
+
+	// Attempt graceful server close
+	doneCh <- struct{}{}
+	<-closedCh
```

Comment:
Can we make `server.Serve` write to `doneCh` on exit instead of adding a `closedCh` here?